Deployment Guides/Provisioning-Islandora-on-Ubuntu-16.md

```diff
@@ -0,0 +1,1796 @@
+
+chown -R blazegraph:blazegraph /usr/share/tomcat7-blzg/
+
+systemctl restart blazegraph
+
+cd ~
+
+rm -rf islandora_log_config
+```
+
+#### Drupal Filter <a id="drupal-filter"></a>
+
+Setup Drupal filter:  
+```
+cd $CATALINA_HOME/webapps/fedora/WEB-INF/lib 
+
+wget --no-check-certificate $DRUPAL_FILTER_URL 
+
+cat > $FEDORA_HOME/server/config/jass.conf <<END_DF
+fedora-auth
+{
+	org.fcrepo.server.security.jaas.auth.module.XmlUsersFileModule required
+	debug=true;
+	ca.upei.roblib.fedora.servletfilter.DrupalAuthModule required
+	debug=true;
+};
+
+fedora-auth-xmlusersfile
+{
+	org.fcrepo.server.security.jaas.auth.module.XmlUsersFileModule required
+	debug=true;
+};
+
+fedora-auth-ldap-bind
+{
+	org.fcrepo.server.security.jaas.auth.module.LdapModule required
+	host.url="ldap://dev01.muradora.org"
+	auth.type="simple"
+	bind.mode="bind"
+	bind.filter="uid={0},ou=people,dc=muradora,dc=org"
+	debug=true;
+};
+
+fedora-auth-ldap-bind-search-bind
+{
+	org.fcrepo.server.security.jaas.auth.module.LdapModule required
+	host.url="ldap://dev01.muradora.org"
+	auth.type="simple"
+	bind.mode="bind-search-bind"
+	bind.user="uid=binduser,ou=people,dc=muradora,dc=org"
+	bind.pass="murabind"
+	search.base="ou=people,dc=muradora,dc=org"
+	search.filter="(uid={0})"
+	attrs.fetch="cn,sn,mail,displayName,carLicense"
+	debug=true;
+};
+
+fedora-auth-ldap-bind-search-compare
+{
+	org.fcrepo.server.security.jaas.auth.module.LdapModule required
+	host.url="ldap://dev01.muradora.org"
+	auth.type="simple"
+	bind.mode="bind-search-compare"
+	bind.user="uid=binduser,ou=people,dc=muradora,dc=org"
+	bind.pass="murabind"
+	search.base="ou=people,dc=muradora,dc=org"
+	search.filter="(uid={0})"
+	attrs.fetch="cn,sn,mail,displayName,carLicense"
+	debug=true;
+};
+END_DF
+
+cd $FEDORA_HOME/server/config  
+
+wget https://raw.github.com/Islandora/islandora_drupal_filter/master/filter-drupal.xml 
+
+sed -i "s|DB_SERVER|$DB_SERVER|g" filter-drupal.xml
+
+sed -i "s|DRUPAL_DB_NAME|$DRUPAL_DB_NAME|g" filter-drupal.xml
+
+sed -i "s|DRUPAL_DB_USER|$DRUPAL_DB_USER|g" filter-drupal.xml 
+
+sed -i "s|DRUPAL_DB_PASS|$DRUPAL_DB_PASS|g" filter-drupal.xml
+```
+
+Start Fedora fully configured:
+
+`chown -R $FEDORA_USER:$FEDORA_USER $FEDORA_HOME`  
+
+`chown -R $FEDORA_USER:$FEDORA_USER $CATALINA_HOME` 
+
+`systemctl restart tomcat7`
+
+### Install Drupal  <a id="install-drupal"></a>
+
+#### CLUI Config <a id="clui-config"></a>
+
+**Note:** Some libraries, modules, contrib are optional but are included to support additional theming. e.g. features, strongarm, node_export.
+ 
+```
+cd $OS_DEFAULT_DOCUMENTROOT
+
+drush dl drupal  
+
+mv drupal-7* drupal7
+
+# or, alternatively (makes upgrades easier)
+#ln -s drupal-7* drupal7
+
+cd drupal7
+
+mkdir sites/default/files  
+
+mkdir sites/all/{modules,themes,libraries}
+
+cp sites/default/default.settings.php sites/default/settings.php
+```
+
+#### Islandora Modules  <a id="islandora-modules"></a>
+
+Git clone the following modules:  
+
+`
```

Comment:
Generally, I would recommend checking out https://github.com/discoverygarden/islandora_gsearcher over the multithreaded gsearch. I have found that it causes issues with stack stability and not worth the headache.

Reply:
Installed now, but not configured.